app/models/test_run.rb

```diff
@@ -12,23 +12,33 @@ class TestRun < ApplicationRecord
   STATUSES = {
     complete: 'complete',
     in_progress: 'in_progress',
-    cancelled: 'cancelled'
+    cancelled: 'cancelled',
+    complete_with_errors: 'complete_with_errors'
   }.freeze
 
   def status
-    return STATUSES[:complete] unless test_script_results.empty?
+    unless test_script_results.empty?
+      if test_script_results.joins(:test_results).where('test_results.completion_status': 'error').count&.positive?
```

Comment:
Hmm can't you use `exists?` or `any?` here?

Reply:
I like `empty?` because it returns false for an array of null values but you're not going to get that with an active record relation. I'll change it.